api/app.rb

```diff
@@ -0,0 +1,111 @@
+require 'sinatra'
+require "sinatra/cross_origin"
+require 'byebug'
+require 'date'
+require 'restclient'
+require 'json'
+
+configure do
+  enable :cross_origin
+end
+
+before do
+  response.headers['Access-Control-Allow-Origin'] = '*'
+end
+
+def endpoint_call(base, month)
+  begin
+    # url = "https://fi7661d6o4.execute-api.eu-central-1.amazonaws.com/prod/indexes/#{params[:base_year]}/#{params[:current_month]}/"
+    # print url
+    url = "https://fi7661d6o4.execute-api.eu-central-1.amazonaws.com/prod/indexes/#{base}/#{month}"
+    response = RestClient.get url
+    intermediary_response = JSON.parse(response)
+    # puts pretty_response = JSON.pretty_generate(intermediary_response)
+    intermediary_response.keys
+    base_index = intermediary_response["index"]["MS_HLTH_IDX"]
+    return base_index
+  rescue RestClient::ExceptionWithResponse => e
+    p e.response
+  end
+end
+
+def removing_month(date_object)
+  date_year = date_object.year
+  date_mon = date_object.mon
+  date_day = date_object.day
+  if date_mon == 1
+    date_mon = 12
+    date_year -= 1
+  else
+    date_mon -= 1
+  end
+  Date.new(date_year, date_mon, date_day)
+end
+
+# ROUTES ...
+post '/v1/indexations' do
+  data = JSON.parse request.body.read
+  signed_on = Date.parse(data["signed"])
+  signed_on = removing_month(signed_on)
+  base_month = signed_on.strftime("%Y-%m")
+
+  start_on = Date.parse(data["start"])
+  start_month = start_on.mon
+  start_day = start_on.day
+
+  present_year = Date.today.year
+  recent_birthday = Date.new(present_year, start_month, start_day)
+  today = Date.today
+
+  if today > recent_birthday
+
+    recent_birthday = removing_month(recent_birthday)
+    current_month = recent_birthday.strftime("%Y-%m")
+  else
+    recent_month = recent_birthday.month - 1
+    recent_day = recent_birthday.day
+    recent_year = Date.today.year - 1
+    current_month = Date.new(recent_year, recent_month, recent_day).strftime("%Y-%m")
+  end
+
+  bases = [1988, 1996, 2004, 2013]
+
+  year_selected = nil
+  bases.map do |year|
+    if year <= Date.parse(data["signed"]).year
+      year_selected = year
+    end
+  end
```

Comment:
Of course! Makes sense, I see I need to learn to go through my code immediately once I am done with testing and refactoring to give the straight answer, rather than keeping in the code that allowed me to "see better" during testing